### PR TITLE
Update service registration code

### PR DESCRIPTION
**Pull request by rynowak**

### CODE REVIEW

File: src/Microsoft.AspNet.Mvc.Core/MvcCoreServiceCollectionExtensions.cs [hunk 1]
```diff
@@ -48,59 +46,49 @@ internal static void AddMvcCoreServices(IServiceCollection services)
...
                 ServiceDescriptor.Transient<IActionDescriptorProvider, ControllerActionDescriptorProvider>());
-            services.TryAdd(ServiceDescriptor
-                .Singleton<IActionDescriptorsCollectionProvider, DefaultActionDescriptorsCollectionProvider>());
+            services.TryAddSingleton<IActionDescriptorsCollectionProvider, DefaultActionDescriptorsCollectionProvider>();
 
             // Action Selection
             //
-            services.TryAdd(ServiceDescriptor.Singleton<IActionSelector, DefaultActionSelector>());
+            services.TryAddSingleton<IActionSelector, DefaultActionSelector>();
             // Performs caching
```
Comment (dougbu) [inline src/Microsoft.AspNet.Mvc.Core/MvcCoreServiceCollectionExtensions.cs]: Getting rather cramped. Please add blank lines to make these blocks more readable.

Comment (dougbu) [inline src/Microsoft.AspNet.Mvc.Core/MvcCoreServiceCollectionExtensions.cs]: Hmm, was packed already. But worth fixing.

File: src/Microsoft.AspNet.Mvc.Core/MvcCoreServiceCollectionExtensions.cs [hunk 2]
```diff
@@ -153,8 +120,9 @@ private static void ConfigureDefaultServices(IServiceCollection services)
             services.AddOptions();
             services.AddRouting();
             services.AddNotifier();
-            services.Configure<RouteOptions>(
-                routeOptions => routeOptions.ConstraintMap.Add("exists", typeof(KnownRouteValueConstraint)));
+
+            services.TryAddEnumerable(
+                ServiceDescriptor.Transient<IConfigureOptions<RouteOptions>, MvcRouteOptionsSetup>());
```
Comment (dougbu) [inline src/Microsoft.AspNet.Mvc.Core/MvcCoreServiceCollectionExtensions.cs]: What is the advantage of the new `MvcRouteOptionsSetup` class over adding the `DefaultFrameworkSortOrder` to previous `Configure<T>()` call?

**Comment by dougbu** [comment]
:watch:

File: src/Microsoft.AspNet.Mvc.Core/MvcCoreServiceCollectionExtensions.cs (hunk 1 above)
Comment (rynowak) [inline src/Microsoft.AspNet.Mvc.Core/MvcCoreServiceCollectionExtensions.cs]: sure

File: src/Microsoft.AspNet.Mvc.Core/MvcCoreServiceCollectionExtensions.cs (hunk 2 above)
Comment (rynowak) [inline src/Microsoft.AspNet.Mvc.Core/MvcCoreServiceCollectionExtensions.cs]: You can use `TryAddEnumerable` to make it idempotent. This is critical for *MVC Core* services.

**Comment by rynowak** [comment]
showed @dougbu updated changes in person.

**Comment by dougbu** [comment]
Yuppy :shipit:

**Pull request closed without merging**